drivers/hmis/app/models/hmis/hud/project.rb

```diff
@@ -234,6 +234,15 @@ def data_collection_features
     end.compact
   end
 
+  def external_form_submissions
```

Comment:
I'm not sure if my understanding is all correct here, but I think perhaps using `detect_best_instance_for_project` here isn't right, because it returns only one instance's worth of form submissions, but we want this to return all the form submissions for the project -- and so we can then filter them from the graphql query depending on which form type the user selects in the dropdown @gigxz  mentioned [here](https://www.figma.com/file/F9EHR2IZYuMAlOJ9aOG2IL?node-id=1879:12521&mode=design#724436105).

this is a snippet of code that I got working locally, though as usual, I'm not sure it's airtight from an arel perspective:
```
  def external_form_submissions
    definition_identifiers = Hmis::Form::Instance.
      for_project(self).
      with_role(:EXTERNAL_FORM).
      active.order(updated_at: :desc).
      pluck(:definition_identifier).
      uniq

    return Hmis::HmisExternalApis::ExternalForms::FormSubmission.none if definition_identifiers.empty?

    HmisExternalApis::ExternalForms::FormSubmission.
      joins(:definition).
      where(definition: { identifier: definition_identifiers })
  end
```

Reply:
That looks correct to me. I think based on the design, we are able to _require_ the definition when querying for the Form Submissions, which would make this a bit simpler. Like:

```
  def external_form_submissions(definition_identifier:)
    # do some validation to make sure the definition identifier is valid for the project
    raise "not valid for the project" unless Hmis::Form::Instance.for_project(self).with_role(:EXTERNAL_FORM).with_identifier(definition_identifier).exists?

    HmisExternalApis::ExternalForms::FormSubmission.joins(:definition).where(fd_t[:identifier].eq(definition_identifier))
  end
```

If we go that route, the frontend would already have the `definition_identifier` because they selected it from the dropdown. Assuming that the `PickListOption` returns something like `{ label: "Prevention Screening Form", code: <definition identifier> }`.

Reply:
Thanks for the improvement and simplification, I'll make that update

Reply:
Updated the resolver and added a required formDefinitionIdentifier argument